src/components/PostData/PostDetails.tsx

```diff
@@ -0,0 +1,85 @@
+import React, {
+  useCallback, useContext, useEffect, useState,
+} from 'react';
+import { NewCommentForm } from '../Comment/NewCommentForm';
+import { DispatchContext, StateContext } from '../../reducer/store';
+import { Loader } from '../Loader';
+import { CommentsList } from '../Comment/CommentsList';
+import { postService } from '../../services/post.service';
+import { Notifications } from '../Notifications/Notifications';
+
+export const PostDetails: React.FC = () => {
+  const [showSpinner, setShowSpinner] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [showCommentForm, setShowCommentForm] = useState(false);
+
+  const { selectedPost, commentsList } = useContext(StateContext);
+  const dispatch = useContext(DispatchContext);
+
+  const onShowFormHandler = useCallback(() => {
+    setShowCommentForm(true);
+  }, []);
+
+  const showNoComment = !isError && !commentsList && !showSpinner;
+  const showCommentsList = commentsList?.length && !showSpinner;
+  const showWriteBtn = !showCommentForm && !showSpinner && !isError;
+
+  useEffect(() => {
+    const { setPostComments } = postService(dispatch);
+
+    setPostComments(
+      selectedPost,
+      setShowSpinner,
+      setIsError,
+    );
+
+    return () => {
+      setIsError(false);
+      setShowCommentForm(false);
+    };
+  }, [selectedPost]);
+
+  return (
+    <div className="content" data-cy="PostDetails">
+      <div className="content" data-cy="PostDetails">
```

Comment:
You have two nested div elements with the same className 'content' and the same 'data-cy' attribute 'PostDetails'. This could lead to confusion and potential styling or testing issues.